kedro/contrib/io/networkx/networkx_local.py

```diff
@@ -0,0 +1,101 @@
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""``NetworkXLocalDataSet`` loads and saves graphs to a local json file in node/link format using
+``NetworkX``.
+See https://networkx.github.io/documentation/stable/tutorial.html for details.
+"""
+
+import json
+from pathlib import Path
+from typing import Any, Dict, Union
+
+import networkx
+
+from kedro.contrib.io import DefaultArgumentsMixIn
+from kedro.io import AbstractDataSet
+
+
+class NetworkXLocalDataSet(DefaultArgumentsMixIn, AbstractDataSet):
+    """
+    ``NetworkXLocalDataSet`` loads and saves graphs to a local JSON file in node/link format using
+    ``NetworkX``.
+    See https://networkx.github.io/documentation/stable/tutorial.html for details.
+
+    Example:
+    ::
+        >>> from kedro.contrib.io.networkx import NetworkXLocalDataSet
+        >>> import networkx as nx
+        >>> graph = nx.complete_graph(100)
+        >>> graph_dataset = NetworkXLocalDataSet(filepath="test.json")
+        >>> graph_dataset.save(graph)
+        >>> reloaded = graph_dataset.load()
+        >>> assert nx.is_isomorphic(graph, reloaded)
+    """
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(
+            filepath=self._filepath,
+            load_args=self._load_args,
+            save_args=self._save_args,
+        )
+
+    def __init__(
+        self,
+        filepath: str,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``NetworkXLocalDataSet``.
+        Args:
+            filepath: The path to the NetworkX graph JSON file.
+            load_args: Arguments passed on to ```networkx.node_link_graph``.
+                See the details in
+                https://networkx.github.io/documentation/networkx-1.9.1/reference/generated/networkx.readwrite.json_graph.node_link_graph.html
+            save_args: Arguments passed on to ```networkx.node_link_data``.
+                See the details in
+                https://networkx.github.io/documentation/networkx-1.9.1/reference/generated/networkx.readwrite.json_graph.node_link_data.html
+
+        """
+        self._filepath = Path(filepath)
+        super().__init__(load_args, save_args)
+
+    def _load(self) -> networkx.Graph:
+        json_payload = json.loads(self._filepath.read_text())
+        graph = networkx.node_link_graph(json_payload, **self._load_args)
+        return graph
+
+    def _save(self, data: networkx.Graph) -> None:
+        json_graph = networkx.node_link_data(data, **self._save_args)
+        with self._filepath.open("w") as output_file:
+            json_payload = json.dumps(json_graph)
```

Comment:
Just `json.dump(json_graph, output_file)` should do the same, with an added benefit of using less memory while processing, because instead of making a massive buffer, it will be writing to the file in chunks.
The same can be done for reading, and for other datasets where possible, but let's start with this one :)